src/modules/preview-site/components/preview-site-row.tsx

```diff
@@ -35,6 +36,19 @@ export function PreviewSiteRow( {
 	const { url, date, isDeleting } = snapshot;
 	const { countDown, expireDateString, isExpired } = useExpirationDate( date );
 	const { fetchSnapshotUsage, removeSnapshot } = useSnapshots();
+	const {
+		progress: deletionProgress,
+		startProgress: startDeletionProgress,
+		stopProgress: stopDeletionProgress,
+	} = useProgress();
+
+	useEffect( () => {
+		if ( isDeleting ) {
+			startDeletionProgress();
+		} else {
+			stopDeletionProgress();
+		}
+	}, [ isDeleting, startDeletionProgress, stopDeletionProgress ] );
```

Comment:
We are calling startDeletion multiple times. I suggest refactoring to its own component.

<img width="192" alt="Screenshot 2025-02-21 at 10 55 32" src="https://github.com/user-attachments/assets/8843b6fd-a388-476a-9967-a4733da0e3c2" />

Reply:
I've created a separate component https://github.com/Automattic/studio/pull/960/commits/6492a251c30c708c68d22b9aba06747abdc11f1e